### PR TITLE
Cancel OSE builds when koji task is cancelled Mk.2

### PR DESCRIPTION
**Pull request by vrutkovs**
Use SIGINT signal handler to cancel OSE build (if it has been started). This change 
ensures that the build in New/Pending state (where logs are not being written) is 
also correctly cancelled.

Tested on local koji instance

### CODE REVIEW

**Comment by vrutkovs** [comment]
PTAL @twaugh @lcarva @pbabinca

**Comment by lcarva** [comment]
Verified that this works as expected.

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 1]
```diff
@@ -406,11 +409,16 @@ def handler(self, src, target_info, arch, scratch=False,
         build_id = build_response.get_build_name()
         self.logger.debug("OSBS build id: %r", build_id)
 
-        self.logger.debug("Waiting for osbs build_id: %s to be scheduled.",
-                          build_id)
-        # we need to wait for kubelet to schedule the build, otherwise it's 500
-        self.osbs().wait_for_build_to_get_scheduled(build_id)
```
Comment (twaugh) [inline koji_containerbuild/plugins/builder_containerbuild.py]: Are we no longer waiting for the build to get scheduled?

Comment (vrutkovs) [inline koji_containerbuild/plugins/builder_containerbuild.py]: Its no longer necessary - OSE doesn't throw HTTP 500 when logs are fetched for Pending build

Comment (twaugh) [inline koji_containerbuild/plugins/builder_containerbuild.py]: I wonder if Origin behaves the same way. Don't want to tie this to OSE specifically.

Comment (vrutkovs) [inline koji_containerbuild/plugins/builder_containerbuild.py]: Didn't try this on Origin, so not sure which version is required for that to work. I'll revert that part, and make sure that `signal.signal` solution works anyway

**Comment by twaugh** [comment]
LGTM from code inspection but I haven't tested it.

**Pull request merged**